src/gui/editors/PianoRoll.cpp

```diff
@@ -3533,7 +3533,8 @@ void PianoRoll::selectAll()
 
 	for( const Note *note : m_pattern->notes() )
 	{
-		int len_ticks = note->length();
+		int len_temp = note->length();
+		int len_ticks = len_temp > 0 ? len_temp : 1;
 
```

Comment:
the first line I add is to cast to an `int`. Should this be a static cast instead?
It would look something like:

```
int len_ticks = static_cast<int>( note->length() ) > 0 ?
				static_cast<int>( note->length() ) : 1;
```